pkg/extkingpin/flags.go

```diff
@@ -43,11 +43,14 @@ func RegisterGRPCFlags(cmd FlagClause) (
 }
 
 // RegisterCommonObjStoreFlags register flags commonly used to configure http servers with.
-func RegisterHTTPFlags(cmd FlagClause) (httpBindAddr *string, httpGracePeriod *model.Duration) {
+func RegisterHTTPFlags(cmd FlagClause) (httpBindAddr *string, httpGracePeriod *model.Duration, httpTLSConfig *string) {
 	httpBindAddr = cmd.Flag("http-address", "Listen host:port for HTTP endpoints.").Default("0.0.0.0:10902").String()
 	httpGracePeriod = ModelDuration(cmd.Flag("http-grace-period", "Time to wait after an interrupt received for HTTP Server.").Default("2m")) // by default it's the same as query.timeout.
-
-	return httpBindAddr, httpGracePeriod
+	httpTLSConfig = cmd.Flag(
+		"http-tls-config",
```

Comment:
One nit, I heard from @roidelapluie that Prometheus flag is called `web.config.file` and not tls-config.file for a good reason: It's not only TLS. I think it's good idea to change it to `http.config` maybe? Also we can expand (contribute) to exporter toolking to support passing parsed configuration too if we want pathorcontent flag (up to us)

Reply:
From what I understand you do not need changes in the downstream exporter toolkit to achieve what you want.

You could probably to all of this in thanos. --http-config={file}  or --http-config=/path:{file} could be supported without downstram changes.

Reply:
Maybe I do not understand the change. Are you trying to use basic auth on certain paths only? Or am I misunderstanding?

Reply:
There is a reason why we take a path. The file is read on every request to make sure that the TLS certificate and the users are up to date.

Reply:
Oh, okay. Then I guess using `pathorcontent` isn't suitable and the normal `Flag` method would be better. Also, no change is needed in exporter or Thanos then. Thanks a lot for explaining! 🙂